iree/compiler/Dialect/HAL/IR/HALOps.td

```diff
@@ -1170,25 +1170,25 @@ def HAL_CommandBufferBindDescriptorSetOp :
   ];
 }
 
-def HAL_CommandBufferDispatchOp : HAL_Op<"command_buffer.dispatch"> {
-  let summary = [{command buffer dispatch recording operation}];
+def HAL_CommandBufferDispatchSymbolOp : HAL_Op<"command_buffer.dispatch.symbol"> {
+  let summary = [{command buffer dispatch recording operation, using symbolref}];
   let description = [{
-    Dispatches an execution request.
+    Dispatches an execution request, using a symbol reference to the entry point.
 
     ```mlir
     %x = constant 128 : index
     %y = constant 32 : index
     %z = constant 1 : index
     hal.command_buffer.dispatch %cmd, %executable,
-                                entry_point = 0,
+                                entry_point = @executable::@target::@entry0,
                                 workgroup_xyz = [%x, %y, %z]
     ```
   }];
 
   let arguments = (ins
     HAL_CommandBuffer:$command_buffer,
-    HAL_Executable:$executable,
-    HAL_OrdinalAttr:$entry_point,
+    HAL_Executable:$executable,  // TODO(scotttodd): remove this and extract from nested ref?
```

Comment:
yeah we can kill this, which is nice (less chance for the two to get out of sync when we end up moving the entry points around - changing the entry point from @foo::@bar -> @taco::@beef is just a single op, not having to also update executable)

Reply:
we can also remove the `entry_point` here, as there will just be one symbol reference

this way, we have this:
`hal.command_buffer.dispatch.symbol %cmd, @executable::@entry_point, ...`
and
`hal.command_buffer.dispatch %cmd, @executable, entry_point=0`

Reply:
@benvanik I'm sorry, what?

Reply:
Oops, `@` tagged you by username, sorry about that! `@foo::@bar` here are symbols >_>
Ben, use code blocks for those next time :P

Reply:
hahaha sorry about that :P

Reply:
Chatted offline a bit. Going to investigate that in a follow-up. It's not as straightforward as removing `HAL_Executable:$executable` here and pulling it out of the symbol reference later. It needs a `hal.executable.lookup` with a `HAL_Device` (which we might be able to carry through the `DispatchState` struct), or something similar.